From: Leon Romanovsky <leon@leon.nu>
Date: Sun, 21 Feb 2016 18:12:26 +0200
Subject: IB/core: Fix missed clean call in registration path
Patch-mainline: v4.5-rc7
Git-commit: 5adebafb75bdfbbe4ec69f14c3613e70f6ed7f6f
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

In case of failure returned from query function in
IB device registration, we need to clean IB cache which
was missed.

This change fixes it.

Fixes: 3e153a93a1c1 ('IB/core: Save the device attributes on the device
structure')
Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
Reviewed-by: Ira Weiny <ira.weiny@intel.com>
Reviewed-by: Sagi Grimberg <sagig@mellanox.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/infiniband/core/device.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/core/device.c
+++ b/drivers/infiniband/core/device.c
@@ -358,6 +358,7 @@ int ib_register_device(struct ib_device
 	ret = device->query_device(device, &device->attrs, &uhw);
 	if (ret) {
 		printk(KERN_WARNING "Couldn't query the device attributes\n");
+		ib_cache_cleanup_one(device);
 		goto out;
 	}
 
